src/prommis/nanofiltration/diafiltration_two_salt.py

```diff
@@ -0,0 +1,1097 @@
+        """
+        super().build()
+
+        # TODO: generalize to any 2 cations and 1 anion
+
+        self.add_mutable_parameters()
+        self.add_variables()
+        self.add_constraints()
+        self.discretize_model()
+        self.add_scaling_factors()
+
+    def add_mutable_parameters(self):
+        """
+        Adds parameters for the two salt diafiltration unit model.
+
+        Assigns default values that can be changed by the user during implementation.
+        """
+        self.membrane_thickness = Param(
+            initialize=1e-7,
+            mutable=True,
+            units=units.m,
+            doc="Thickness of membrane (z-direction)",
+        )
+        self.membrane_width = Param(
+            initialize=1,
+            mutable=True,
+            units=units.m,
+            doc="Width of the membrane (x-direction)",
+        )
+        self.membrane_length = Param(
+            initialize=100,
+            mutable=True,
+            units=units.m,
+            doc="Length of the membrane, wound radially",
+        )
+        self.applied_pressure = Param(
+            initialize=10,
+            mutable=True,
+            units=units.bar,
+            doc="Pressure applied to membrane",
+        )
+        self.membrane_permeability = Param(
+            initialize=0.01,
+            mutable=True,
+            units=units.m / units.h / units.bar,
+            doc="Hydraulic permeability coefficient",
+        )
+        self.feed_flow_volume = Param(
+            initialize=100,
+            mutable=True,
+            units=units.m**3 / units.h,
+            doc="Volumetric flow rate of the feed",
+        )
+        self.feed_conc_mass_lithium = Param(
+            initialize=1.7,
+            mutable=True,
+            units=units.kg / units.m**3,
+            doc="Mass concentration of lithium in the feed",
+        )
+        self.feed_conc_mass_cobalt = Param(
+            initialize=17,
+            mutable=True,
+            units=units.kg / units.m**3,
+            doc="Mass concentration of cobalt in the feed",
+        )
+        self.diafiltrate_flow_volume = Param(
+            initialize=30,
+            mutable=True,
+            units=units.m**3 / units.h,
+            doc="Volumetric flow rate of the diafiltrate",
+        )
+        self.diafiltrate_conc_mass_lithium = Param(
+            initialize=0.1,
+            mutable=True,
+            units=units.kg / units.m**3,
+            doc="Mass concentration of lithium in the diafiltrate",
+        )
+        self.diafiltrate_conc_mass_cobalt = Param(
+            initialize=0.2,
+            mutable=True,
+            units=units.kg / units.m**3,
+            doc="Mass concentration of cobalt in the diafiltrate",
+        )
+        self.temperature = Param(
+            initialize=298,
+            mutable=True,
+            units=units.K,
+            doc="System temperature",
+        )
```

Comment:
Is this going to be optimized? Then it should be a `Var`. In the long run, it might be good to put it on the property package, but, since no energy balance exists, we should avoid that at present.